nemo/collections/nlp/models/language_modeling/megatron_gpt_prompt_learning_model.py

```diff
@@ -353,7 +386,44 @@ def validation_step(self, batch, batch_idx):
         if loss_mean.item == 0.0:
             loss_mean = []
 
-        return loss_mean
+        if self.cfg.get('report_validation_metric', False):
+            preds_text, labels_text = [], []
+            input_ids, labels, loss_mask, position_ids, attention_mask, taskname_ids = batch
+            input_lenghts = torch.argmax(loss_mask, 1, keepdim=True)
+
+            res = megatron_gpt_generate(
+                self.cuda(),
+                (
```

Comment:
Can we split this out into multiple lines for readability? Preferably also explain to people why we need to concatenate torch.zeros --> "it's because megatron_gpt_generate expects the input_ids to contain tokens for it to overwrite instead of appending to